mteb/tasks/Classification/eng/LinguisticProbing.py

```diff
@@ -0,0 +1,454 @@
+from __future__ import annotations
+
+from mteb.abstasks import AbsTaskClassification
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class SentLenClassification(AbsTaskClassification):
+    metadata = TaskMetadata(
+        name="SentLenClassification",
+        description="This is a classification task where the goal is to predict the sentence length which has been binned in 6 possible categories with lengths ranging in the following intervals: --0: (5-8), 1: (9-12), 2: (13-16), 3: (17-20), 4: (21-25), 5: (26-28).",
+        reference="https://arxiv.org/abs/1805.01070",
+        dataset={
+            "path": "tasksource/linguisticprobing",
+            "revision": "1372f0696870b8ad09ec3ed123173be722c7a891",
+            "name": "sentence_length",
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=("2018-01-01", "2018-07-08"),
+        form=["written"],
+        domains=["Non-fiction", "Fiction"],
+        task_subtypes=["Topic classification"],
+        license="BSD",
+        socioeconomic_status="mixed",
+        annotations_creators="expert-annotated",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation="""
+        @misc{conneau2018cram,
+      title={What you can cram into a single vector: Probing sentence embeddings for linguistic properties}, 
+      author={Alexis Conneau and German Kruszewski and Guillaume Lample and Loïc Barrault and Marco Baroni},
+      year={2018},
+      eprint={1805.01070},
+      archivePrefix={arXiv},
+      primaryClass={cs.CL}
+        }
+        """,
+        n_samples={"test": 8507},
+        avg_character_length={"test": 86.80},
+    )
+
+    def dataset_transform(self):
+        self.dataset = self.dataset.rename_column("sentence", "text")
+        self.dataset = self.stratified_subsampling(
+            self.dataset, seed=self.seed, splits=["test"]
+        )
+
+
+class WCClassification(AbsTaskClassification):
```

Comment:
Correct me if I'm wrong but this does 1000-way classification, and my guess would be that it would take a lot of time and examples to train a classification head. Perhaps a keyword retrieval task would do the same trick way faster.

Reply:
Probably, it would work better but I stick with the paper's choice. The better option is to treat the probing task as a new dataset and we can change whatever we want

Reply:
Hmmm, I think from a validity standpoint of view it's fine to stick with the paper's choice.
I'm just a bit worried that it would take a very long time to run.